composables/useCustomerCreditCards.ts

```diff
@@ -0,0 +1,151 @@
+    } finally {
+      loading.value = false
+    }
+  }
+
+  const updatePaymentCardDetails = async (accountId, cardId, cardInput) => {
+    loading.value = true
+    try {
+      const variables = {
+        accountId,
+        cardId,
+        cardInput,
+      }
+
+      const response = await fetcher({
+        query: updatePaymentCard,
+        variables,
+      })
+      console.log(response)
+      // await load(cards?.value?.items?.)
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.error(err)
+    } finally {
+      loading.value = false
+    }
+  }
+
+  const deletePaymentCardDetails = async (accountId, cardId) => {
+    loading.value = true
+    try {
+      const variables = {
+        accountId,
+        cardId,
+      }
+
+      const response = await fetcher({
+        query: deletePaymentCard,
+        variables,
+      })
+      console.log(response)
+      // await load(cards?.value?.items?.)
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.error(err)
+    } finally {
+      loading.value = false
+    }
+  }
+
+  const load = async (accountId: Number) => await getPaymentCardDetails(accountId)
+
+  const updateCard = async (accountId: Number, cardId: String, cardInput: CardInput) =>
+    await updatePaymentCardDetails(accountId, cardId, cardInput)
+
+  const deleteCard = async (accountId: Number, cardId: String) =>
+    await deletePaymentCardDetails(accountId, cardId)
+
+  const tokenizeCreditCard = async (creditCardData) => {
+    // creditCardData, { pciHost, tenantId }
+    try {
+      const url = `https://${pciHost}/payments/commerce/payments/cards/`
+      const tenantId = apiHost.split("//")[1].split("-")[0].split("t")[1].toString()
+      const { cardNumber, cardType, cvv } = creditCardData
+      const ccData = { cardNumber, cardType, cvv }
+
+      const res = await fetch(url, {
+        method: "POST",
+        headers: {
+          accept: "application/json",
+          "x-vol-tenant": tenantId,
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(ccData),
+      })
+      const tokenizedCCData = await res.json()
+      if (tokenizedCCData.isSuccessful) {
+        return tokenizedCCData
+      }
+    } catch (e) {
+      console.error(e)
+    }
+  }
+
+  const addPaymentMethod = async (orderId, paymentAction) => {
+    try {
+      const variables = {
+        orderId,
+        paymentAction,
+      }
+
+      const response = await fetcher({
+        query: addPaymentCard,
+        variables,
+      })
+      console.log(response)
+      // await load(cards?.value?.items?.)
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.error(err)
+    } finally {
+      loading.value = false
+    }
+  }
+
+  const tokenizeCard = async (creditCardData) => {
+    return await tokenizeCreditCard(creditCardData)
+  }
+
+  const addPaymentMethodByTokenizeCard = async (orderId, paymentAction) => {
```

Comment:
There is a mix of customer account and checkout card actions in this composable.

The addPaymentMethodByTokenizeCard and other checkout actions should be moved out

Reply:
Created usePaymentMethod composable for addPaymentMethod handling.